src/pages/Ranking/index.js

```diff
@@ -14,73 +15,108 @@ import './styles.css';
 const WORD_HEIGHT = 40;
 
 const Ranking = () => {
-  const { currentUser, currentUserPlayed, dailyResults, expandedUser, setExpandedUser } =
-    useRankingData();
+  const { push } = useHistory();
+
+  const {
+    currentUser,
+    currentUserPlayed,
+    currentStreakRankingData,
+    dailyResults,
+    expandedUser,
+    setExpandedUser,
+    loading,
+  } = useRankingData();
 
   return (
     <div className="ranking">
       <h1 className="section-title">Ranking</h1>
-      <Tabs className="Tabs">
-        <TabList>
-          <Tab>Today</Tab>
-          <Tab>All Time Average</Tab>
-        </TabList>
-        <TabPanel>
-          {dailyResults.map(
-            ({ attemptedWords, attempts, position, status, user: { email, name, photo } }) => {
-              const isCurrentUser = email === currentUser;
-              const lost = status === GAME_STATUS.lost;
-              const isExpanded = email === expandedUser;
-              const isDisabled = !currentUserPlayed;
+      {loading ? (
```

Comment:
I would suggest refactoring this page and splitting the responsibilities between pages and components. With the current approach, you can potentially re-render the entire page on any props change. It can produce performance issues. As an example you can leave the main page layout at this level.

```
<div className="ranking">
      <h1 className="section-title">Ranking</h1>
     <RankingList ... />
</div>
```

Inside RankingList you can make something like

```
...
if (loading) return <Loading />

return         
<Tabs className="Tabs">
      <TabList>
            <Tab>{`Today (${dailyResults.length})`}</Tab>
            <Tab>{`Current Streak (${currentStreakRankingData.length})`}</Tab>
      </TabList>
      <TabPanel>
            {dailyResults.map(dailyResult => <DailyResultDailyResultRow ... />)}
      </TabPanel>
      <TabPanel>
            {dailyResults.map(currentStreak => <RankingListCurrentStreakRow ... />)}
      </TabPanel>
</Tabs>
```

Reply:
awesome, thanks! Will do this change in another PR dedicating to refactoring